ddtrace/settings.py

```diff
@@ -115,5 +118,134 @@ def __repr__(self):
         return '{}.{}({})'.format(cls.__module__, cls.__name__, keys)
 
 
+class Hooks(object):
+    """
+    Hooks configuration object is used for registering and calling hook functions
+
+    Example::
+
+        @config.falcon.hooks.request
+        def on_request(span, request, response):
+            pass
+    """
+    __slots__ = ['_hooks']
+
+    def __init__(self):
+        # DEV: We have a custom `__setattr__`, call the inherited `object.__setattr__`
+        super(Hooks, self).__setattr__('_hooks', collections.defaultdict(set))
+
+    def __getattr__(self, key):
+        """
+        Return a hook decorator for the attribute name provided
+
+        Example::
+
+            @config.falcon.hooks.request
+            def on_request(span, request, response):
+                pass
+
+
+        This is the same as::
+
+            @config.falcon.hooks.register('request')
+            def on_request(span, request, response):
+                pass
+
+        :param key: Hook name to register
+        :type key: str
+        :returns: A function decorator used to register a hook
+        :rtype: function
+        """
+        return self.register(key)
```

Comment:
There is some ergonomic benefit added here. However I think this makes things less clear in the long run, especially if we begin adding other methods on `Hooks`. (What if we want to support a hook for something called  `'deregister'`?)